pinot-tools/src/main/java/org/apache/pinot/tools/tuner/TunerCommand.java

```diff
@@ -0,0 +1,102 @@
+  private long _numQueriesToGiveRecommendation = DEFAULT_NUM_QUERIES_TO_GIVE_RECOMMENDATION;
+
+  @Option(name = "-tables", required = false, usage = "Comma separated list of table names to work on without type (unset run on all tables)")
+  private String _tableNamesWithoutType = null;
+
+  @Option(name = "-help", required = false, help = true, aliases = {"-h", "--h", "--help"}, usage = "Print this message.")
+  private boolean _help;
+
+  @Override
+  public boolean execute() {
+    HashSet<String> tableNamesWithoutType = new HashSet<>();
+    if (_tableNamesWithoutType != null && !_tableNamesWithoutType.trim().equals("")) {
+      tableNamesWithoutType.addAll(Arrays.asList(_tableNamesWithoutType.split(",")));
+    }
+
+    if (_strategy.equals(INVERTED_INDEX)) {
+      TunerDriver parserBased = new TunerDriver().setThreadPoolSize(Runtime.getRuntime().availableProcessors() - 1)
+          .setStrategy(new ParserBasedImpl.Builder()
+              .setTableNamesWithoutType(tableNamesWithoutType)
+              .setNumProcessedThreshold(_numQueriesToGiveRecommendation)
+              .setAlgorithmOrder(ParserBasedImpl.FIRST_ORDER)
+              .setNumEntriesScannedThreshold(_numEntriesScannedThreshold)
+              .build())
+          .setQuerySrc(new LogQuerySrcImpl.Builder()
+              .setParser(new BrokerLogParserImpl())
+              .setPath(_brokerLog)
+              .build())
+          .setMetaManager(new JsonFileMetaManagerImpl.Builder()
+              .setPath(_metaData)
+              .build());
+      parserBased.execute();
+    } else if (_strategy.equals(SORTED_INDEX)) {
+      TunerDriver parserBased = new TunerDriver().setThreadPoolSize(Runtime.getRuntime().availableProcessors() - 1)
+          .setStrategy(new ParserBasedImpl.Builder()
+              .setTableNamesWithoutType(tableNamesWithoutType)
+              .setNumProcessedThreshold(_numQueriesToGiveRecommendation)
+              .setAlgorithmOrder(ParserBasedImpl.SECOND_ORDER)
+              .setNumEntriesScannedThreshold(_numEntriesScannedThreshold)
+              .build())
+          .setQuerySrc(new LogQuerySrcImpl.Builder()
+              .setParser(new BrokerLogParserImpl())
+              .setPath(_brokerLog)
+              .build())
+          .setMetaManager(new JsonFileMetaManagerImpl.Builder()
+              .setPath(_metaData)
+              .build());
+      parserBased.execute();
+    } else {
+      return false;
+    }
+    return true;
+  }
+
+  @Override
+  public String description() {
+    return "Give optimization boundary analysis and indexing recommendation to specific tables, based on packed segment metadata (containing the weighted sum of cardinality, number of documents, number of entries, etc.) and broker logs (containing time of execution, numEntriesScannedInFilter, numEntriesScannedPostFilter, query text).";
```

Comment:
Not sure what "give optimization boundary analysis" means.. can you make this bit simpler?